### PR TITLE
LPS-129113

### PR DESCRIPTION
**Pull request by joshuacords**
https://issues.liferay.com/browse/LPS-129113

Author: @joshuacords 

cc: @lipusz

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
(I'll push a fixed SF branch once testing is running)

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by lipusz** [comment]
Closing for now until we are investigating this problem further.

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/DefaultKeywordQueryContributor.java [hunk 1]
```diff
@@ -76,13 +83,47 @@ public void contribute(
 		}
 	}
 
+	public int getMaximumKeywordSize() {
+		return _maximumKeywordSize;
+	}
+
+	public String trimKeywords(String keywords) {
+		if (keywords.length() < _maximumKeywordSize) {
```
Comment (lipusz) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/DefaultKeywordQueryContributor.java:91]: Null check :bug:

File: portal-kernel/src/com/liferay/portal/kernel/search/SearchContext.java [hunk 2]
```diff
@@ -366,7 +368,13 @@ public void setIncludeStagingGroups(boolean includeStagingGroups) {
 	}
 
 	public void setKeywords(String keywords) {
-		_keywords = keywords;
+		if (_defaultKeywordQueryContributorBridge != null) {
```
Comment (lipusz) [inline portal-kernel/src/com/liferay/portal/kernel/search/SearchContext.java:371]: :eyes:

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/DefaultKeywordQueryContributor.java (hunk 1 above)
Comment (lipusz) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/DefaultKeywordQueryContributor.java:91]: Also, we probably not want to limit the number of keywords, just the length of an individual keyword so this check is not needed.

Comment (joshuacords) [reply modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/DefaultKeywordQueryContributor.java:91]: For the vast majority of cases we expect search terms to be below the _maximumKeywordSize, and String manipulation is somewhat expensive, I think a fast return here is worthwhile.

**Pull request closed without merging**